src/main/scala-2.11/org/apache/james/gatling/smtp/SmtpAction.scala

```diff
@@ -1,70 +1,54 @@
 package org.apache.james.gatling.smtp
 
-import io.gatling.commons.stats.{KO, OK}
-import io.gatling.core.Predef._
+import akka.actor.{Actor, Props}
 import io.gatling.core.action.{Action, ChainableAction}
 import io.gatling.core.session.Session
 import io.gatling.core.stats.StatsEngine
-import io.gatling.core.stats.message.ResponseTimings
-import org.apache.commons.mail.SimpleEmail
 import org.apache.james.gatling.control.UserFeeder
 
-import scala.util.{Failure, Success, Try}
+import scala.util.Failure
 
-class SmtpAction(
-                  requestName: String,
+object SmtpAction {
+  def props(requestName: String, subject: String, body: String, statsEngine: StatsEngine, next: Action, protocol: SmtpProtocol) =
+    Props(new SmtpAction(requestName, subject, body, statsEngine, next, protocol))
+}
+
+class SmtpAction(requestName: String,
                   subject: String,
                   body: String,
                   val statsEngine: StatsEngine,
                   val next: Action,
-                  protocol: SmtpProtocol) extends ChainableAction {
+                  protocol: SmtpProtocol) extends ChainableAction with Actor {
 
   val name = "sendMail"
 
   def execute(session: Session) {
-    val email: SimpleEmail = generateEmail(session)
-    val reqStart = System.currentTimeMillis()
-    val executionReport = sendMail(email)
-    val reqEnd = System.currentTimeMillis()
-
-    statsEngine.logResponse(session, requestName, ResponseTimings(reqStart, reqEnd), executionReport.status, None, executionReport.errorMessage)
+    def readSession(variableName: String): String = session.apply(variableName).as[String]
```

Comment:
`def readSession(variableName: String) = session(variableName).as[String]` is enough IMO